cocos/2d/assembler/sprite/simple.ts

```diff
@@ -170,6 +142,42 @@ export const simple: IAssembler = {
             vData[27] = ar + cttx;
             vData[28] = br + dtty;
         }
+    },
+
+    fillBuffers (sprite: Sprite, renderer: Batcher2D) {
+        if (sprite === null) {
+            return;
+        }
+
+        const vData = sprite.renderData!.vData!;
+        if (sprite.node.hasChangedFlags || (this.uiTransCache && this.uiTransCache._rectDirty)) {
```

Comment:
We can only check for `sprite.node.hasChangedFlags` here. UITransform size and anchor dirty should be updated directly in `updateVertexData`